subscribe.go

```diff
@@ -153,12 +153,18 @@ func sendHeaders(w http.ResponseWriter, s *Subscriber) {
 }
 
 // retrieveLastEventID extracts the Last-Event-ID from the corresponding HTTP header with a fallback on the query parameter.
-func retrieveLastEventID(r *http.Request) string {
+func retrieveLastEventID(r *http.Request, logger Logger) string {
 	if id := r.Header.Get("Last-Event-ID"); id != "" {
 		return id
 	}
 
-	return r.URL.Query().Get("Last-Event-ID")
+	if id := r.URL.Query().Get("Last-Event-ID"); id != "" {
```

Comment:
nit: IMHO we should check if the key exists, even if it contains an empty string, to trigger the deprecation (JS code may always include this query parameter even if empty).